### DISCUSSION TITLE
json.dumps({"r": range(0,16,5)})

### DISCUSSION THREAD

**Discussion by Beormund** [micropython/micropython#10244] [Using MicroPython] [answered]
```py
json.dumps({"r": range(0,16,5)})
# >>> '{"r": range(0, 16, 5)}'
```
Should this not throw an error? It generates invalid json.

Using Peter Hinch's Scheduling module I would like to load/save schedules from/to json.  In Python "every 5 minutes" is represented as mins = range(0, 60, 5). 

```py
asyncio.create_task(schedule(foo, 'every 5 mins', hrs=None, mins=range(0, 60, 5)))
```

Any ideas how to represent this as json? I thought maybe "mins": {"start": 0, "stop": 60, "step": 5} which is quite verbose but easy to transform the dict to range.

**Comment by glenn20** [comment, marked as answer]
This might be what you were looking for:
```python
json.dumps({"r": list(range(0,16,5))})
```
-> `'{"r": [0, 5, 10, 15]}'`

I agree it would be better if micropython raised a TypeError (like cpython does).

**Reply by peterhinch** [reply]
That would be my approach. The requirement is for "An object supporting the Python iterator protocol and iterating over integers." (from the docs). So you can present the list directly to the `mins` arg.
